### ISSUE TITLE
Wear media controls do not reflect playback state

### ISSUE THREAD

**Issue by mchowning** [Automattic/pocket-casts-android#1050]
### Description

Default media notification and media controls for the watch app are not reflecting playback state from Pocket Casts watch app.

### Step-by-step reproduction instructions

1. Launch the pocket casts watch app and begin playback
2. 👎  Observe that there is no playing notification on the watch home screen
3. 👎 Observe that the watch media controls app indicates that nothing is playing

### Screenshots or screen recording

| | Watch home screen | Watch media controls |
| --- | --- | --- |
| Playing on PC watch app | ![image](https://github.com/Automattic/pocket-casts-android/assets/4656348/b866ee19-da96-450c-857f-25e3a1abc48e) | ![image](https://github.com/Automattic/pocket-casts-android/assets/4656348/b10f5aec-791e-4ab4-960e-428801e47475) |
| Playing on PC phone app | ![media-icon](https://github.com/Automattic/pocket-casts-android/assets/4656348/3363806d-c48c-46c8-8994-3e8b18747884) | ![image](https://github.com/Automattic/pocket-casts-android/assets/4656348/79460de7-0e4f-4321-9a93-e5e674786413) |

### Did you search for existing bug reports?

- [X] I have searched for existing bug reports.

### Device, Operating system, and Pocket Casts app version

Observed on both a Galaxy Watch 4 and a Pixel watch. 7.40 pre-release build

**Comment by mchowning** [comment]
Turns out that this did work previously on the emulator at least.

### Removing a patron icon fixes the issue on the emulator, sort of...

If you build 4d38c45af and run it on an emulator, when I play an episode I get a message saying that the wear app has crashed. The app continues to play the episode, however, but the wear media controls do not reflect that there is any playback. If I update that commit with only these changes, however (essentially reverting a portion of the changes in that commit), then the wear os does not crash and the media controls accurately reflect playback on the emulator (but still not on a physical watch).


<details><summary>Code diff</summary>


``` diff
diff --git a/modules/features/settings/src/main/java/au/com/shiftyjelly/pocketcasts/settings/AppearanceIconSettingsAdapter.kt b/modules/features/settings/src/main/java/au/com/shiftyjelly/pocketcasts/settings/AppearanceIconSettingsAdapter.kt
index b3b0c3813..d37ed351d 100644
--- a/modules/features/settings/src/main/java/au/com/shiftyjelly/pocketcasts/settings/AppearanceIconSettingsAdapter.kt
+++ b/modules/features/settings/src/main/java/au/com/shiftyjelly/pocketcasts/settings/AppearanceIconSettingsAdapter.kt
@@ -92,7 +92,7 @@ class AppearanceIconSettingsAdapter(
             if (showOption && selected) {
                 tickDrawable = AppCompatResources.getDrawable(itemView.context, R.drawable.ic_circle_tick)
             } else if (!showOption) {
-                tickDrawable = AppCompatResources.getDrawable(itemView.context, R.drawable.ic_locked_plus)
+                tickDrawable = AppCompatResources.getDrawable(itemView.context, R.drawable.ic_plus_bubble)
             }
             binding.imgTick.setImageDrawable(tickDrawable)
             binding.imgTick.contentDescription = binding.imgTick.resources.getString(if (selected) LR.string.on else LR.string.off)
diff --git a/modules/features/settings/src/main/java/au/com/shiftyjelly/pocketcasts/settings/AppearanceThemeSettingsAdapter.kt b/modules/features/settings/src/main/java/au/com/shiftyjelly/pocketcasts/settings/AppearanceThemeSettingsAdapter.kt
index c9cda60c3..b1e577ed6 100644
--- a/modules/features/settings/src/main/java/au/com/shiftyjelly/pocketcasts/settings/AppearanceThemeSettingsAdapter.kt
+++ b/modules/features/settings/src/main/java/au/com/shiftyjelly/pocketcasts/settings/AppearanceThemeSettingsAdapter.kt
@@ -93,7 +93,7 @@ class AppearanceThemeSettingsAdapter(
             if (showOption && selected) {
                 tickDrawable = AppCompatResources.getDrawable(itemView.context, R.drawable.ic_circle_tick)
             } else if (!showOption) {
-                tickDrawable = AppCompatResources.getDrawable(itemView.context, R.drawable.ic_locked_plus)
+                tickDrawable = AppCompatResources.getDrawable(itemView.context, R.drawable.ic_plus_bubble)
             }
             binding.imgTick.setImageDrawable(tickDrawable)
             binding.imgTick.contentDescription = binding.imgTick.resources.getString(if (selected) LR.string.on else LR.string.off)
diff --git a/modules/features/settings/src/main/res/drawable/ic_locked_plus.xml b/modules/features/settings/src/main/res/drawable/ic_locked_plus.xml
deleted file mode 100644
index e619596ad..000000000
--- a/modules/features/settings/src/main/res/drawable/ic_locked_plus.xml
+++ /dev/null
@@ -1,25 +0,0 @@
-<vector xmlns:android="http://schemas.android.com/apk/res/android"
-    xmlns:aapt="http://schemas.android.com/aapt"
-    android:width="40dp"
-    android:height="40dp"
-    android:viewportWidth="40"
-    android:viewportHeight="40">
-  <path
-      android:pathData="M20.117,20.117m-13.749,-0.118a13.75,13.75 124.03,1 1,27.499 0.235a13.75,13.75 124.03,1 1,-27.499 -0.235">
-    <aapt:attr name="android:fillColor">
-      <gradient 
-          android:startX="8.941"
-          android:startY="6.271"
-          android:endX="33.794"
-          android:endY="8.635"
-          android:type="linear">
-        <item android:offset="0" android:color="#FFFED745"/>
-        <item android:offset="1" android:color="#FFFEB525"/>
-      </gradient>
-    </aapt:attr>
-  </path>
-  <path
-      android:pathData="M16.75,18.031V17.5C16.75,15.705 18.205,14.25 20,14.25C21.795,14.25 23.25,15.705 23.25,17.5V18.031C23.681,18.142 24,18.534 24,19V24C24,24.552 23.552,25 23,25H17C16.448,25 16,24.552 16,24V19C16,18.534 16.319,18.142 16.75,18.031ZM18.25,17.5C18.25,16.534 19.034,15.75 20,15.75C20.966,15.75 21.75,16.534 21.75,17.5V18H18.25V17.5Z"
-      android:fillColor="#ffffff"
-      android:fillType="evenOdd"/>
-</vector>
```

</details> 


I don't even think any of that code is directly used by the wear app. 🤔 

Applying these changes to the current head commit on `main` (172f07fb9), does _not_ fix the issue (even on the emulator). That may mean there is another change somewhere along the line that is also causing this issue independently. I'm not sure how much time I want to put into tracking that down since it's not clear that is getting us any close to getting the media notification working on an actual physical device since applying the diff above only addresses the issue on an emulator.

### Wear 4.0

If we run the current head commit on `main ` (172f07fb9) on a emulator running the Wear OS 4.0 preview (instead of my regular emulator running wear 3.5), then the media notification shows up fine without making any changes to the code.

**Comment by mchowning** [comment]
I was able to narrow down the kind of diff that can cause this issue. It appears to be related to the gradient in this icon because removing the gradient brings back the notification on wear:

``` diff
diff --git a/modules/features/settings/src/main/res/drawable/ic_locked_plus.xml b/modules/features/settings/src/main/res/drawable/ic_locked_plus.xml
index e619596ad..234711543 100644
--- a/modules/features/settings/src/main/res/drawable/ic_locked_plus.xml
+++ b/modules/features/settings/src/main/res/drawable/ic_locked_plus.xml
@@ -5,18 +5,8 @@
     android:viewportWidth="40"
     android:viewportHeight="40">
   <path
-      android:pathData="M20.117,20.117m-13.749,-0.118a13.75,13.75 124.03,1 1,27.499 0.235a13.75,13.75 124.03,1 1,-27.499 -0.235">
-    <aapt:attr name="android:fillColor">
-      <gradient 
-          android:startX="8.941"
-          android:startY="6.271"
-          android:endX="33.794"
-          android:endY="8.635"
-          android:type="linear">
-        <item android:offset="0" android:color="#FFFED745"/>
-        <item android:offset="1" android:color="#FFFEB525"/>
-      </gradient>
-    </aapt:attr>
+      android:pathData="M20.117,20.117m-13.749,-0.118a13.75,13.75 124.03,1 1,27.499 0.235a13.75,13.75 124.03,1 1,-27.499 -0.235"
+      android:fillColor="#FFFED745">
   </path>
   <path
       android:pathData="M16.75,18.031V17.5C16.75,15.705 18.205,14.25 20,14.25C21.795,14.25 23.25,15.705 23.25,17.5V18.031C23.681,18.142 24,18.534 24,19V24C24,24.552 23.552,25 23,25H17C16.448,25 16,24.552 16,24V19C16,18.534 16.319,18.142 16.75,18.031ZM18.25,17.5C18.25,16.534 19.034,15.75 20,15.75C20.966,15.75 21.75,16.534 21.75,17.5V18H18.25V17.5Z"
```

I don't think the gradient itself is the problem though because I can replace this file with another file that we already have in the app that has gradients (i.e., [ic_create_new_account](https://github.com/Automattic/pocket-casts-android/blob/d1502d9bf89fec3e19098f3ec78bc41e1473b447/modules/features/account/src/main/res/drawable/ic_create_new_account.xml#L21)), and the same problem happens. 

Even more bizarrely, if I take the "fixed" version of "ic_locked_plus" (i.e., with the diff above applied) and copy that to be the content of "ic_create_new_account", the notification stops working, but if I copy it to `ic_done` instead, then the notification keeps working.

**Comment by mchowning** [comment]
Did some more digging on this, and got some more info.

The TLDR is that there is a bitmap read exception that Wear OS throws when the media notification does not work. I am able to avoid that exception on the emulator, and when I do that the media notification works. Unfortunately,  on a physical watch I haven't found a way to avoid the exception and get the media notification to work. To be clear, I don't know with complete certainty that this exception is what causes the media notification to not work, but it's looking likely.

The most interesting finding from this is that adding an icon file can break the media notification on the emulator even if that icon file is completely unused in the app.

 I'm using the [try/wear-notification-bitmap-issue](https://github.com/Automattic/pocket-casts-android/compare/main...try/wear-notification-bitmap-issue) branch to explore this a bit and try to isolate the issue. I made the changes in a specific order of commits to show what is happening, so let me describe what each commit does and what effect it has on the issue.

1. That branch starts from https://github.com/Automattic/pocket-casts-android/commit/4d38c45af60ad7c5b3fc4d07124d5511a1e333e6, which is the commit identified earlier where the media notification stopped working on the emulator.

2. **_Fixing the media notification:_** In f697a3bd8f8c538748c8fe6fabfae4f644c14f44, I have removed the icon that was causing the issue on the emulator, so the media notification is working again on the emulator with this commit.

3. **_Avoiding an irrelevant caught exception that clutters the logs_**: In ca503c0b4fd2d22e768734623e62980949bee377, I am just avoiding an unrelated exception getting printed in the logs because that confuses the issues.

4. **_Using a super-simple media notification_**: In c77aca7fb1439b144831792b896a4bdd90fa7827, I simplified the media notification to the bare minimum that is needed to demonstrate the bitmap exception crash within Wear OS in order to avoid any complexity from it. This media notification only has the session id attached to it and a small icon. At this point, the media notification is working on the emulator. It is not working on the physical watch though (because it never works on the physical watch for me), and the physical watch os throws the following StreamLooper exception:

```
FATAL EXCEPTION: StreamLooper
             Process: com.google.android.wearable.app, PID: 13593
             java.lang.NullPointerException: Attempt to invoke virtual method 'int android.graphics.Bitmap.getByteCount()' on a null object reference
```

5. **_Adding an unused icon breaks the media notification_**: In 574daca3fa607048ee0eb5dba67413dd6b12ac51, adding an icon with a gradient **that is not used at all** breaks the media notification on the emulator, and the emulator now throws the same bitmap exception that the physical watch throws. If I take [the media notification we're building](https://github.com/Automattic/pocket-casts-android/commit/eb51b245a2d4f9f57362c6915204e30287f0f4da#diff-45c6e9da6b23edc6bf0f4f105eecbfd6842691bc17533396f76774bc80e037eaR77-R78) and remove the small icon, then the bitmap exception is no longer thrown and the media notification works on the emulator.

6. **_Removing gradient from unused icon "fixes" the media notification_**: In 40935964643ada716ea2655e0705c5000558594e, I comment out the portion of the icon XML that creates a gradient. This "fixes" the issue on the emulator (no bitmap exception and media notification works), but the physical watch still throws the exception and the media notification does not work there.

In order to isolate the cause of the bitmap exception on the physical watch I tried removing all of the XML `<aapt:attr ....>` elements from the app, but the physical watch still threw the bitmap exception and the media notification did not work.

In addition, it does not appear that this specific gradient is causing the problem because I put the same xml element creating that gradient in a different file in the app, and the media notification continued to work on the emulator, but when I put it in another file in another file in the app, it broke the media notification. So the same xml gradient can break the media notification or not depending on what file I add it to. 😕
